src/datasets/arrow_dataset.py

```diff
@@ -3353,6 +3354,421 @@ def init_buffer_and_writer():
         else:
             yield rank, True, shard
 
+    def reduce(
+        self,
+        function: Optional[Callable] = None,
+        input_columns: Optional[Union[str, List[str]]] = None,
+        initializer: Optional[Union[Any, List[Any]]] = None,
+        batched: bool = False,
+        batch_size: Optional[int] = 1000,
+        drop_last_batch: bool = False,
+        keep_in_memory: bool = False,
+        load_from_cache_file: bool = None,
+        cache_file_name: Optional[str] = None,
+        fn_kwargs: Optional[dict] = None,
+        num_proc: Optional[int] = None,
+        new_fingerprint: Optional[str] = None,
+        desc: Optional[str] = None,
+    ) -> Dict[str, Any]:
+        """
+        Reduce the examples in the table (individually or in batches) using a binary operation and return the result.
+
+        The function should be a binary operation, i.e. it takes in 2 objects of the same type in and returns 1 object of the same type as the inputs.
+        Examples include addition, multiplication, maximum, minimum, etc. for `int`, `float`, `bool` and `str` types, and concatenation for `list` types.
+        The first input is the accumulant, which is the result of the previous application of the function on the previous examples, and the second input is the current example.
```

Comment:
(nit) I'd call this "accumulator" here and in the rest of the code

```suggestion
        The first input is the accumulator, which is the result of the previous application of the function on the previous examples, and the second input is the current example.
```

Reply:
Good point, ill get on it asap.